bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -1,62 +1,65 @@
+import {input} from "core/dom"
 import * as p from "core/properties"
 import {Widget, WidgetView} from "models/widgets/widget"
 
 export class FileInputView extends WidgetView {
   model: FileInput
 
-  protected dialogEl: HTMLInputElement
+  protected dialog_el: HTMLInputElement
 
   connect_signals(): void {
     super.connect_signals()
     this.connect(this.model.change, () => this.render())
-    this.connect(this.model.properties.width.change, () => this.render())
   }
 
   render(): void {
-    if (this.dialogEl == null) {
-      this.dialogEl = document.createElement('input')
-      this.dialogEl.type = "file"
-      this.dialogEl.multiple = this.model.multiple
-      this.dialogEl.onchange = () => {
-        const {files} = this.dialogEl
+    const {multiple, accept, disabled, width} = this.model
+
+    if (this.dialog_el == null) {
+      this.dialog_el = input({type: "file", multiple})
+      this.dialog_el.onchange = () => {
+        const {files} = this.dialog_el
         if (files != null) {
           this.load_files(files)
         }
       }
-      this.el.appendChild(this.dialogEl)
+      this.el.appendChild(this.dialog_el)
+    }
+
+    if (accept != null && accept != "") {
+      this.dialog_el.accept = accept
     }
-    if (this.model.accept != null && this.model.accept != '')
-      this.dialogEl.accept = this.model.accept
 
-    this.dialogEl.style.width = `{this.model.width}px`
-    this.dialogEl.disabled = this.model.disabled
+    this.dialog_el.style.width = `${width}px`
+    this.dialog_el.disabled = disabled
   }
 
   async load_files(files: FileList): Promise<void> {
-    const value: string[] = []
-    const filename: string[] = []
-    const mime_type: string[] = []
-    let i: number
-
-    for (i = 0; i < files.length; i++){
-      filename.push(files[i].name)
-      const data_url = await this.readfile(files[i])
-      const [, mime, , data] = data_url.split(/[:;,]/, 4)
-      value.push(data)
-      mime_type.push(mime)
+    const values: string[] = []
+    const filenames: string[] = []
+    const mime_types: string[] = []
+
+    for (const file of files) {
+      const data_url = await this._read_file(file)
+      const [, mime_type="",, value=""] = data_url.split(/[:;,]/, 4)
+
+      values.push(value)
```

Comment:
How much sense does it make to leave the value as base64 encoded? It would be good to change this, but that would be a solid backwards incompatible change.

Reply:
I don't have a strong opinion except that if you want to change that, 3.0 is the place.

Reply:
and to make an issue to track it and also so that we can telegraph it as an upcoming change